### PR TITLE
Updating protobuf for v1.5

### PR DESCRIPTION
**Pull request by flagxor**
Bringing this in sync with the latest changes and clarifications for v1.5

Adds:
* Report Type
* Days since onset
* revised_keys to list revoked / revised keys

### CODE REVIEW

**Comment by sethvargo** [comment]
```
internal/pb/export/export.proto:77:8: Option "(nanopb)" unknown. Ensure that your proto definition file imports the proto which defines the option.
```

@flagxor

**Comment by cjwagner** [comment]
/test pull-en-server-release-unit-canary

(Please ignore, testing out a change to the presubmit.)

File: internal/pb/export/export.proto [hunk 1]
```diff
@@ -41,21 +41,30 @@ message TemporaryExposureKeyExport {
...
+  // TemporaryExposureKeys that have changed status.
+  // Keys should be included in this list if they have changed status
+  // or have been revoked.
+  repeated TemporaryExposureKey revised_keys = 8;
 }
 
 message SignatureInfo {
-  // No longer need to set these. Old code that still does will be fine
-  // as these were largely noops anyway.
+  // These fields are no longer needed (confirmed with Apple)
```
Comment (gurayAlsac) [inline internal/pb/export/export.proto:56]: I'd revert this comment. That these are confirmed with Apple is sorta implied all throughout.

File: internal/pb/export/export.proto [hunk 2]
```diff
@@ -67,24 +76,44 @@ message TemporaryExposureKey {
...
   optional int32 rolling_start_interval_number = 3;
 
   // Increments of 10 minutes describing how long a key is valid
-  optional int32 rolling_period = 4
-      [default = 144];  // defaults to 24 hours
+  optional int32 rolling_period = 4 [default = 144];  // defaults to 24 hours
+
+  // Data type representing why this key was published.
+  enum ReportType {
+    UNKNOWN = 0;  // Never returned by the API.
```
Comment (gurayAlsac) [inline internal/pb/export/export.proto:90]: optional nit: we might leave out the "// Never returned by the API" comments in this repo, which is server focused.

File: internal/pb/export/export.proto [hunk 3]
```diff
@@ -67,24 +76,44 @@ message TemporaryExposureKey {
...
 }
 
 message TEKSignature {
-  // Info about the signing key, version, algorithm, etc
+  // Info about the signing key, version, algorithm, etc. Only the
+  // verification_key_id, verification_key_version, and
+  // signature_algorithm fields within signature_info are read.
   optional SignatureInfo signature_info = 1;
-  // E.g., Batch 2 of 10
+  // E.g., Batch 2 of 10 - these fields are ignored on android in favor of the
```
Comment (gurayAlsac) [inline internal/pb/export/export.proto:115]: I'd revert these comments as well and leave our implementation some flexibility.

File: internal/pb/export/export.proto (hunk 1 above)
Comment (flagxor) [reply internal/pb/export/export.proto:56]: done

**Comment by sethvargo** [comment]
Gonna defer the lgtm to @gurayAlsac

File: internal/pb/export/export.proto (hunk 3 above)
Comment (flagxor) [reply internal/pb/export/export.proto:115]: done

File: internal/pb/export/export.proto (hunk 2 above)
Comment (flagxor) [reply internal/pb/export/export.proto:90]: I'd rather leave this in as each of these comments got meticulously covered with Apple.
(And less diff to manage between copies)

File: internal/pb/export/export.proto [hunk 4]
```diff
@@ -67,22 +75,41 @@ message TemporaryExposureKey {
...
 
 message TEKSignatureList {
   repeated TEKSignature signatures = 1;
 }
 
 message TEKSignature {
-  // Info about the signing key, version, algorithm, etc
+  // Info about the signing key, version, algorithm, etc. Only the
+  // verification_key_id, verification_key_version, and
+  // signature_algorithm fields within signature_info are read.
```
Comment (gurayAlsac) [inline internal/pb/export/export.proto:112]: Same

**Comment by gurayAlsac** [comment]
> Can you file an issue to support this? We need a redesign to handle keys that can change values.

Filed https://github.com/google/exposure-notifications-server/issues/662

**Comment by flagxor** [comment]
> Can you file an issue to support this? We need a redesign to handle keys that can change values.

https://github.com/google/exposure-notifications-server/issues/663

File: internal/pb/export/export.proto [hunk 5]
```diff
@@ -55,7 +62,8 @@ message SignatureInfo {
   optional string verification_key_version = 3;
   // Alias with which to identify public key to be used for verification
   // Must be in character class [a-zA-Z0-9_]
-  // For cross-compatibility with Apple, use MCC (https://en.wikipedia.org/wiki/Mobile_country_code)
+  // For cross-compatibility with Apple, use MCC
```
Comment (mikehelmick) [inline internal/pb/export/export.proto:65]: @gurayAlsac I thought we were dropping MCC requirements?

File: internal/pb/export/export.proto (hunk 2 above)
Comment (mikehelmick) [reply internal/pb/export/export.proto:89]: Are we guaranteeing that the server will never publish a key w/ UNKNOWN?

File: internal/pb/export/export.proto [hunk 6]
```diff
@@ -67,22 +75,41 @@ message TemporaryExposureKey {
...
+  optional int32 rolling_period = 4 [default = 144];  // defaults to 24 hours
+
+  // Data type representing why this key was published.
+  enum ReportType {
+    UNKNOWN = 0;  // Never returned by the API.
+    CONFIRMED_TEST = 1;
+    CONFIRMED_CLINICAL_DIAGNOSIS = 2;
+    SELF_REPORT = 3;
+    RECURSIVE = 4;
+    REVOKED = 5;  // Used to revoke a key, never returned by API.
```
Comment (mikehelmick) [inline internal/pb/export/export.proto:94]: Is this true - if a key is revoked, will we have a status of REVOKE in the revised_keys list?

File: internal/pb/export/export.proto (hunk 2 above)
Comment (flagxor) [reply internal/pb/export/export.proto:89]: There's an affordance to let the client specify how unsupported reportTypes are handled, so I don't think we need to make any particular guarantee for the server.

Comment (mikehelmick) [reply internal/pb/export/export.proto:89]: ok, the opposite of Guray's nit - we should specify that the server will publish keys w/ status UNKNOWN if we receive them.

File: internal/pb/export/export.proto (hunk 5 above)
Comment (gurayAlsac) [reply internal/pb/export/export.proto:65]: Unfortunately only when iOS 13.6 comes out.

File: internal/pb/export/export.proto (hunk 2 above)
Comment (gurayAlsac) [reply internal/pb/export/export.proto:89]: There's a conflation here about which api we're talking about. The api this comment refers to is the EN API on device, right? That's why I was proposing removing the comment here. Or we could specify which api.

File: internal/pb/export/export.proto (hunk 4 above)
Comment (gurayAlsac) [reply internal/pb/export/export.proto:112]: Don't forget about this one. Let's leave implementation flexible.

File: internal/pb/export/export.proto (hunk 5 above)
Comment (flagxor) [reply internal/pb/export/export.proto:65]: Dunno, this was a reformatting change that g4 fix gave me on the flip side.

File: internal/pb/export/export.proto (hunk 6 above)
Comment (flagxor) [reply internal/pb/export/export.proto:94]: If we want the client to learn of the revocation, yep.

**Comment by flagxor** [comment]
> ```
> internal/pb/export/export.proto:77:8: Option "(nanopb)" unknown. Ensure that your proto definition file imports the proto which defines the option.
> ```
> 
> @flagxor

Bucket bailed by mistake.
dropped

File: internal/pb/export/export.proto (hunk 5 above)
Comment (flagxor) [reply internal/pb/export/export.proto:65]: Shall I drop this?

File: internal/pb/export/export.proto (hunk 2 above)
Comment (flagxor) [reply internal/pb/export/export.proto:89]: Clarified.

File: internal/pb/export/export.proto (hunk 4 above)
Comment (flagxor) [reply internal/pb/export/export.proto]: done

File: internal/pb/export/export.proto [hunk 7]
```diff
@@ -67,22 +75,39 @@ message TemporaryExposureKey {
   optional bytes key_data = 1;
 
   // Varying risks associated with exposure depending on type of verification
-  optional int32 transmission_risk_level = 2;
+  // Ignored by the v1.5 client API when report_type is set.
+  optional int32 transmission_risk_level = 2 [deprecated = true];
```
Comment (gurayAlsac) [inline internal/pb/export/export.proto:79]: Is what we refer to as "v1.5" a notion that external devs will have access to? On iOS side this'll be an os version?

Comment (flagxor) [reply internal/pb/export/export.proto:79]: I believe we've been messaging it that way and Apple will do the same.

Comment (flagxor) [reply internal/pb/export/export.proto:79]: I.e. API version 1.5

**Pull request merged**